### ISSUE TITLE
License Concern

### ISSUE THREAD

**Issue by tylerjroach** [begeekmyfriend/yasea#499]
Hi Leo,

I noticed that x264 is used for software encoding. x264 is currently licensed under GPL, which means any works including it must also be GPL and therefore any apps using yasea must be fully open sourced.

Simple RTMP is also licensed as LGPL which is a little more lenient in terms of licensing. As long as any modifications made to yasea are open source, then the library can be added into a project without having to open source the entire app.

Because many users will want hardware encoding only (API 16+), it may be best to create the software encoder as a module that can easily be ripped out.

In the LICENCE it may be best to explain that code that you have contributed is under MIT  but parts of the code are LGPL / GPL.

Let me know what you think of this, and correct me if you believe any of the statements I have made are wrong.

Thanks!

**Comment by begeekmyfriend** [comment]
Good idea. It is really a headache that we have to put different incompatible licenses into one project. I think one compromise is to open a independent `non-gpuimage` branch in which I would ripped out `x264` soft encoding module and give some explains in `README`.

**Comment by tylerjroach** [comment]
Sounds great, I had actually already done this when I first cloned the repo to see how embedded the soft encoder actually was in the project. I can submit a pull request on this tomorrow if you would like.

**Comment by begeekmyfriend** [comment]
Well, It is so kind of you and I am waiting for you!

**Comment by zzugyl** [comment]
openh264 replace for libx264 ？

**Comment by tylerjroach** [comment]
Sorry for the late response. I've been busy with other work. I have stripped x264 in my own fork of YASEA. Not sure if you want to create a separate branch. Let me know if you want me to create a pull request. 
https://github.com/tylerjroach/yasea/commit/78592db6ebae15b997056eb0679830dbc1c6fcb2

**Comment by begeekmyfriend** [comment]
@tylerjroach That is interesting for me. Please issue a PR on another branch forked from the latest master branch if you can.

**Comment by tylerjroach** [comment]
@begeekmyfriend I think you have to create the new branch for me to create the pull request on your repo. If you can do that, I'll submit the PR right away.

**Comment by begeekmyfriend** [comment]
This [branch](https://github.com/begeekmyfriend/yasea/tree/h264) is welcome for your PR.

**Comment by tylerjroach** [comment]
We now have a branch with x264 stripped out. Upon further review, I noticed MagicCamera, which is used doesn't have a license attached to it. https://github.com/wuhaoyu1990/MagicCamera

It looks like some of MagicCamera is based on GPUImage https://github.com/CyberAgent/android-gpuimage.

I just wanted to bring this to your attention. I may try to strip this out for my fork, maybe just removing filters all together. If this interests you, I'll let you know if I have completed this.
